From 02640e8e653354c3f871717c0b3f967e4135ae1f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Fri, 25 Jan 2013 14:23:58 +0100
Subject: [PATCH] set peer for point-to-point connections

network/net-lib.sh: only set net vars, if they have a value
---
 modules.d/40network/ifup.sh    |  4 ++--
 modules.d/40network/net-lib.sh | 10 +++++-----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/modules.d/40network/ifup.sh b/modules.d/40network/ifup.sh
index 83685c8..4216fa6 100755
--- a/modules.d/40network/ifup.sh
+++ b/modules.d/40network/ifup.sh
@@ -113,10 +113,10 @@ do_static() {
     [ -n "$mtu" ] && ip link set mtu $mtu dev $netif
     if strstr $ip '*:*:*'; then
         # note no ip addr flush for ipv6
-        ip addr add $ip/$mask dev $netif
+        ip addr add $ip/$mask ${srv+peer $srv} dev $netif
     else
         ip addr flush dev $netif
-        ip addr add $ip/$mask brd + dev $netif
+        ip addr add $ip/$mask ${srv+peer $srv} brd + dev $netif
     fi
 
     [ -n "$gw" ] && echo ip route add default via $gw dev $netif > /tmp/net.$netif.gw
diff --git a/modules.d/40network/net-lib.sh b/modules.d/40network/net-lib.sh
index d9a241b..4a4d7e7 100644
--- a/modules.d/40network/net-lib.sh
+++ b/modules.d/40network/net-lib.sh
@@ -271,11 +271,11 @@ ip_to_var() {
     case $# in
         0)  autoconf="error" ;;
         1)  autoconf=$1 ;;
-        2)  dev=$1; autoconf=$2 ;;
-        3)  dev=$1; autoconf=$2; mtu=$3 ;;
-        4)  dev=$1; autoconf=$2; mtu=$3; macaddr=$4 ;;
-        *)  ip=$1; srv=$2; gw=$3; mask=$4;
-            hostname=$5; dev=$6; autoconf=$7; mtu=$8;
+        2)  [ -n "$1" ] && dev=$1; [ -n "$2" ] && autoconf=$2 ;;
+        3)  [ -n "$1" ] && dev=$1; [ -n "$2" ] && autoconf=$2; [ -n "$3" ] && mtu=$3 ;;
+        4)  [ -n "$1" ] && dev=$1; [ -n "$2" ] && autoconf=$2; [ -n "$3" ] && mtu=$3; [ -n "$4" ] && macaddr=$4 ;;
+        *)  [ -n "$1" ] && ip=$1; [ -n "$2" ] && srv=$2; [ -n "$3" ] && gw=$3; [ -n "$4" ] && mask=$4;
+            [ -n "$5" ] && hostname=$5; [ -n "$6" ] && dev=$6; [ -n "$7" ] && autoconf=$7; [ -n "$8" ] && mtu=$8;
             if [ -n "${9}" -a -n "${10}" -a -n "${11}" -a -n "${12}" -a -n "${13}" -a -n "${14}" ]; then
                 macaddr="${9}:${10}:${11}:${12}:${13}:${14}"
             fi
